modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/internal/upgrade/v9_1_2/test/ObjectRelationshipUpgradeProcessTest.java

```diff
@@ -0,0 +1,154 @@
+import com.liferay.portal.test.log.LogCapture;
+import com.liferay.portal.test.log.LoggerTestUtil;
+import com.liferay.portal.test.rule.Inject;
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+import com.liferay.portal.upgrade.registry.UpgradeStepRegistrator;
+import com.liferay.portal.upgrade.test.util.UpgradeTestUtil;
+import com.liferay.portal.vulcan.util.LocalizedMapUtil;
+
+import java.util.Arrays;
+
+import org.junit.Assert;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Thalles Montenegro
+ */
+@RunWith(Arquillian.class)
+public class ObjectRelationshipUpgradeProcessTest {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new LiferayIntegrationTestRule();
+
+	@Test
+	public void testUpgradeProcess() throws Exception {
+		Company company1 = CompanyTestUtil.addCompany();
+
+		User user1 = UserTestUtil.addUser(company1);
+
+		ObjectDefinition objectDefinition1 = _publishCustomObjectDefinition(
+			user1.getUserId());
+
+		_userLocalService.deleteUser(user1);
+
+		Company company2 = CompanyTestUtil.addCompany();
+
+		User user2 = UserTestUtil.addUser(company2);
+
+		ObjectDefinition objectDefinition2 = _publishCustomObjectDefinition(
+			user2.getUserId());
+
+		_userLocalService.deleteUser(user2);
+
+		_runUpgrade();
+
+		long defaultServiceAccountUserId =
+			_userLocalService.getUserIdByScreenName(
+				company1.getCompanyId(), "default-service-account");
+
+		_assertObjectDefinitionUserId(
+			defaultServiceAccountUserId,
+			objectDefinition1.getObjectDefinitionId());
+
+		defaultServiceAccountUserId = _userLocalService.getUserIdByScreenName(
+			company2.getCompanyId(), "default-service-account");
+
+		_assertObjectDefinitionUserId(
+			defaultServiceAccountUserId,
+			objectDefinition2.getObjectDefinitionId());
+	}
+
+	private ObjectDefinition _addCustomObjectDefinition(
+			String label, String name, String pluralLabel, long userId)
+		throws Exception {
+
+		return _objectDefinitionLocalService.addCustomObjectDefinition(
+			userId, 0, false, false, false,
+			LocalizedMapUtil.getLocalizedMap(label), name, null, null,
+			LocalizedMapUtil.getLocalizedMap(pluralLabel), true,
+			ObjectDefinitionConstants.SCOPE_COMPANY,
+			ObjectDefinitionConstants.STORAGE_TYPE_DEFAULT,
+			Arrays.asList(
+				ObjectFieldUtil.createObjectField(
+					ObjectFieldConstants.BUSINESS_TYPE_TEXT,
+					ObjectFieldConstants.DB_TYPE_STRING,
+					RandomTestUtil.randomString(), StringUtil.randomId())));
+	}
+
+	private void _assertObjectDefinitionUserId(
+			long expectedUserId, long objectDefinitionId)
+		throws Exception {
+
+		ObjectDefinition objectDefinition =
+			_objectDefinitionLocalService.getObjectDefinition(
+				objectDefinitionId);
+
+		Assert.assertEquals(expectedUserId, objectDefinition.getUserId());
+	}
+
+	private ObjectDefinition _publishCustomObjectDefinition(long userId)
```

Comment:
Better to inline the method `_publishCustomObjectDefinition` also to follow the pattern we have in `ObjectDefinitionLocalServiceTest`